### ISSUE TITLE
Network Explorer: nicer display of graphs

### ISSUE THREAD

**Issue by ajdapretnar** [biolab/orange3-network#276]
##### Network version
1.9.0

##### Orange version
3.37.0

##### Expected behavior
Gephi (and probably other network visualization tools) have a "smart" display of dense networks. When the network is dense, one can decrease the strength of the edges (i.e. by setting a paler color or thinner line). But more importantly, node labels appear only for larger nodes (if size is set, otherwise for more densely connected).

![gephi-lite-export](https://github.com/user-attachments/assets/79df4e1c-99ec-47fc-9d71-4eafc0e0f65b)

##### Actual behavior
Edges in Orange are black, making it impossible to read the (black) labels.
Moreover, all labels are displayed. One can circumvent this by selecting top N largest nodes in a Data Table and passing a data subset, but it would be nice to have this option directly in Network Explorer.
Labels are also displayed quite far from their nodes, making it difficult to establish which label belongs to which node in a large network.
<img width="552" alt="Screenshot 2024-07-30 at 10 59 44" src="https://github.com/user-attachments/assets/5850ce82-d50b-4b44-8b74-1b2ecbccb53c">

##### Steps to reproduce the behavior
Create any large network. Label the nodes.

##### Additional info (worksheets, data, screenshots, ...)
<img width="960" alt="ParlamintGB-net" src="https://github.com/user-attachments/assets/40369e4c-14fe-4b05-b1ea-a26467cf52f9">

##### TLDR;

- [ ] option to set edge color (to lighter)
- [ ] display only top N nodes
- [ ] font size in node labels to correspond to the size of the node?
- [ ] place labels closer to nodes

**Comment by janezd** [comment]
> Gephi (and probably other network visualization tools) have a "smart" display of dense networks. When the network is dense, one can decrease the strength of the edges (i.e. by setting a paler color or thinner line)

You can already set the edge width. We can allow making it thinner.

> Edges in Orange are black, making it impossible to read the (black) labels.

This can be easily fixed. Mind that #249 will also allow different colors for edges.

> Moreover, all labels are displayed. One can circumvent this by selecting top N largest nodes in a Data Table and passing a data subset, but it would be nice to have this option directly in Network Explorer.

You can already check "Label only selection and subset". This doesn't help?

> Labels are also displayed quite far from their nodes, making it difficult to establish which label belongs to which node in a large network.

This is only true for large networks. The current distance look nice for small number of nodes. Maybe we could add some heuristic and draw the labels closer if the network is dense? It would require some (lots of) experimenting.

I agree that this widget could be made much much better. But it requires a dedicated person - preferably one who is constantly using it and can make all the cosmetic touches. But you already know that ... :(

Also, what I fear is that this widget would have too much options and would become an app by itself.
